Scott2bReal/nasty-snacks
# Upgrade to astro 3, add view transitions/CSR

## Description

Upgrades the app to Astro 3 (all deps now at `latest`), `node-version` to `20.5.0`, and adds a few features:

* Optimized images - the few images on the site are now using Astro's [stable `Image` component](https://docs.astro.build/en/guides/images/#image--astroassets). Further optimizations are on the table...
* View transitions - uses Astro's [new `ViewTransitions` component](https://docs.astro.build/en/guides/view-transitions/) to enable client side routing (and fixes some resulting JS bugs)
* Removes element causing sticky points in page scrolling

Got a little carried away with upgrades while working on this and added some project-wide scripts for linting and formatting, so this PR is a bit noisy. Will likely add github actions to run these as checks on PRs in the future.

## Testing
* View on Chrome, navigate between home and /contact to see new page transitions (should fade in/out)
* Navigate on Firefox to make sure fallback looks good (view transitions are not implemented in Firefox yet)
* Confirm animations on homepage fire on initial load & after navigation via client side router
